apps/token/src/routes/home/index.tsx

```diff
@@ -1,147 +1,199 @@
+      if (!validatorsData?.epoch.timestamps.expiry) return;
+      const now = Date.now();
+      const expiry = new Date(validatorsData.epoch.timestamps.expiry).getTime();
+
+      if (now > expiry) {
+        refetch();
+        clearInterval(epochInterval);
+      }
+    }, 10000);
+
+    return () => {
+      clearInterval(epochInterval);
+    };
+  }, [refetch, validatorsData?.epoch.timestamps.expiry]);
+
+  const proposals = useMemo(
+    () =>
+      proposalsData
+        ? getNotRejectedProposals<ProposalFieldsFragment>(
+            proposalsData.proposalsConnection
+          ).slice(0, 3)
+        : [],
+    [proposalsData]
+  );
+
+  const activeNodes = removePaginationWrapper(
+    validatorsData?.nodesConnection.edges
+  );
+
+  const trimmedActiveNodes = activeNodes?.slice(0, nodesToShow);
+
+  const consensusNodes = activeNodes.filter(
+    (node) =>
+      node.rankingScore.status ===
+      Schema.ValidatorStatus.VALIDATOR_NODE_STATUS_TENDERMINT
+  );
 
   return (
-    <>
-      <Heading title={t('pageTitleHome')} />
-      <HomeSection>
-        <TokenDetails
-          totalSupply={appState.totalSupply}
-          totalAssociated={totalAssociated}
-        />
-      </HomeSection>
-      <HomeSection>
-        <SubHeading title={t('Token Vesting')} />
-        <p>
-          {t(
-            'The vesting contract holds VEGA tokens until they have become unlocked.'
-          )}
-        </p>
-        <p>
-          <Trans
-            i18nKey="Tokens are held in different <trancheLink>Tranches</trancheLink>. Each tranche has its own schedule for how the tokens are unlocked."
-            components={{
-              trancheLink: (
-                <Link
-                  data-testid="tranches-link"
-                  to={Routes.TRANCHES}
-                  className="underline text-white"
-                />
-              ),
-            }}
-          />
-        </p>
-        <p>
-          {t(
-            'Once unlocked they can be redeemed from the contract so that you can transfer them between wallets.'
-          )}
-        </p>
-        <Link to={Routes.REDEEM}>
-          <Button
-            variant="primary"
-            size="md"
-            data-testid="check-vesting-page-btn"
-          >
-            {t('Check to see if you can redeem unlocked VEGA tokens')}
-          </Button>
-        </Link>
-      </HomeSection>
-      <HomeSection>
-        <SubHeading title={t('Use your Vega tokens')} />
-        <p>
-          {t(
-            'To use your tokens on the Vega network they need to be associated with a Vega wallet/key.'
-          )}
-        </p>
-        <p>
-          {t(
-            'This can happen both while held in the vesting contract as well as when redeemed.'
+    <AsyncRenderer
+      loading={proposalsLoading || validatorsLoading}
+      error={proposalsError || validatorsError}
+      data={proposalsData && validatorsData}
+    >
+      {proposalsData && (
```

Comment:
how would this be undefined? Don't the conditions in the async renderer gaurentee this won't be the case

Reply:
great point!